### ISSUE TITLE
Problems with Pursuit scenario from MekHQ

### ISSUE THREAD

**Issue by SuperStucco** [MegaMek/megamek#2966]
I've tried the attached 49.2 MM game, generated from MekHQ, several times.  Each time has presented problems.  "PreMod" is as-generated, "PostMod" is after tweaking deployment turn and elevations.

The first time through, despite changing the pursuing forces to deploy on turn 3 they deployed at the start of game, as it was originally set up.  This change was made through multiple select => configure => change deployment turn.  The second time (saved as PostMod) they were individually changed, which seems to have worked.

The second time through, the infantry was skipped every movement phase.  They were left sitting stationary in the deployment zone.  They only had a chance to move once the other team deployed.  Infantry initiative is set to disperse evenly throughout the turn.

[PreMod.sav.gz](https://github.com/MegaMek/megamek/files/6710865/PreMod.sav.gz)
[PostMod.sav.gz](https://github.com/MegaMek/megamek/files/6710866/PostMod.sav.gz)
A couple of custom units are involved.  They can be found zipped here: https://github.com/MegaMek/mekhq/issues/2703#issuecomment-864107984

Short version - mass select deployment turn selection didn't work, infantry not given the choice to move until enemy team deployed.  There's also a serious issue with deployment order for a "pursuit" scenario, with the chasers deploying at the same round or even before the chased forces are deployed, but that's more on MekHQ.
